tests/platforms/order-by.test.ts

```diff
@@ -0,0 +1,121 @@
+import { Entity, IDatabaseDriver, PrimaryKey, Property, QueryOrder, SimpleLogger, Utils } from '@mikro-orm/core';
+import { MikroORM } from '@mikro-orm/core';
+import { mockLogger } from '../helpers';
+import { PLATFORMS } from '../bootstrap';
+
+@Entity()
+class Test {
+
+  @PrimaryKey()
+  id!: number;
+
+  @Property({ length: 255, nullable: true })
+  value?: string | null;
+
+}
+
+const options = {
+  mysql: { port: 3308 },
+  mariadb: { port: 3309 },
+  mssql: { password: 'Root.Root' },
+  postgresql: {},
+};
+
+describe.each(Utils.keys(options))('Order by [%s]', type => {
+  let orm: MikroORM;
+
+  beforeAll(async () => {
+    orm = await MikroORM.init<IDatabaseDriver>({
+      entities: [Test],
+      driver: PLATFORMS[type],
+      dbName: 'order-by',
+      loggerFactory: SimpleLogger.create,
+      ...options[type],
+    });
+
+    await orm.schema.refreshDatabase();
+  });
+
+  beforeEach(() => orm.schema.clearDatabase());
+
+  afterAll(() => orm.close(true));
+
+  test(`order-by`, async () => {
+    const mock = mockLogger(orm, ['query', 'query-params']);
+
+    await orm.em.findAll(Test,
+      {
+        orderBy: {
+          value: QueryOrder.ASC,
+        },
+    });
```

Comment:
all of those should be written like that, i can imagine this wont pass the linter
```suggestion
    await orm.em.findAll(Test, {
      orderBy: {
        value: QueryOrder.ASC,
      },
    });
```